local_planner/src/nodes/common.cpp

```diff
@@ -54,27 +54,42 @@ PolarPoint cartesianToPolar(double x, double y, double z,
 Eigen::Vector2i polarToHistogramIndex(const PolarPoint& p_pol, int res) {
   // TODO change logic here, as 0,0 are valid index
   Eigen::Vector2i ev2(0, 0);
-  float e = p_pol.e;
-  float z = p_pol.z;
-  if (res <= 0.f || e < -90.f || e > 90.f || z < -180.f || z > 180.f) {
-    return ev2;
+  PolarPoint p_wrapped = wrapPolar(p_pol);
+  // elevation angle to y-axis histogram index
+  if (p_wrapped.e == 90.0f) {
+    p_wrapped.e = 89.0f;
   }
-  if (e == 90.f) {
-    e = 89;
+  p_wrapped.e += 90.0f;
+  p_wrapped.e = p_wrapped.e + (res - (static_cast<int>(p_wrapped.e) % res));
+  ev2.y() = p_wrapped.e / res - 1;
+  // azimuth angle to x-axis histogram index
+  if (p_wrapped.z == 180.0f) {
+    p_wrapped.z = -180.0f;
   }
-  e += 90.0;
-  e = e + (res - (static_cast<int>(e) % res));  //[-80,+90]
-  ev2.y() = e / res - 1;
-
-  if (z == 180.f) {
-    z = -180;
-  }
-  z += 180.0;
-  z = z + (res - (static_cast<int>(z) % res));  //[-80,+90]
-  ev2.x() = z / res - 1;
+  p_wrapped.z += 180.0f;
+  p_wrapped.z = p_wrapped.z + (res - (static_cast<int>(p_wrapped.z) % res));
+  ev2.x() = p_wrapped.z / res - 1;
   return ev2;
 }
 
+PolarPoint wrapPolar(PolarPoint p_pol) {
```

Comment:
Not the end of the world, but this could be done in-place, similarly to the `wrapAngleToPlusMinus180()`, aka make this `void wrapPolar(PolarPoint& p_pol){....}`

Reply:
yes, can be done. now a copy of the original polar point is created in the polarToHistogramIndex, as the input polar point should not be modified in the process.